File: Java/3612.java
import java.util.*;
class Solution {
    public String processStr(String s) {
        StringBuilder word = new StringBuilder();
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == '#') word.append(word);
            else if(s.charAt(i) == '*'){
                if(word.length()>0) word.deleteCharAt(word.length()-1);
            }
            else if(s.charAt(i) == '%') word.reverse();
            else word.append(s.charAt(i));
        }
        return String.valueOf(word);
    }
}

File: Python/3612.py
class Solution(object):
    def processStr(self, s):
        lst = []
        for i in s:
            if i == '*':
                if lst:
                    lst.pop()
            elif i == '#':
                lst.extend(lst)
            elif i == '%':
                lst.reverse()
            else:
                lst.append(i)
        return "".join(lst)